app/controllers/items_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ItemsController < ApplicationController
+  def index
+    @items = Item.all
+  end
+
+  def show_by_merchant
+    @merchant = Merchant.find(params[:merchant_id])
+    @items = @merchant.items
+  end
+
+  def show
+    @item = Item.find(params[:id])
+  end
+
+  def new
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
+  def create
+    merchant = Merchant.find(params[:merchant_id])
+    merchant.items.create!(item_params)
```

Comment:
This line is throwing an error from your tests because you don't have a status in those item params. You can either give the status a default value at the database level (in a migration), or you can manually add the status here in the controller. I was able to fix it with:

```
def create
    merchant = Merchant.find(params[:merchant_id])
    new_item_params = item_params
    new_item_params[:status] = true
    merchant.items.create!(new_item_params)
    redirect_to "/merchants/#{merchant.id}/items"
  end
```